src/ServiceControl/SagaAudit/SagaAuditing.cs

```diff
@@ -127,6 +129,33 @@ public override void Enrich(IReadOnlyDictionary<string, string> headers, IDictio
                 }
 
             }
+
+            static IEnumerable<string> SplitInvokedSagas(string sagasInvokedRaw)
+            {
+                var semicolonCount = sagasInvokedRaw.Count(c => c == ';');
+                var colonCount = sagasInvokedRaw.Count(c => c == ':');
+                if (colonCount != semicolonCount + 1) //Malformed data coming from old version of saga audit plugin
+                {
+                    var tailSemicolon = sagasInvokedRaw.LastIndexOf(";", StringComparison.Ordinal);
+                    var tail = sagasInvokedRaw.Substring(tailSemicolon + 1);
+                    var head = sagasInvokedRaw.Substring(0, tailSemicolon);
+
+                    var headDeduplicated = head.Substring(0, head.Length / 2);
```

Comment:
Doesn't this rely on the two sagas being invoked being the same type. What if the incoming message invokes two different sagas?